### PR TITLE
[API] Add repoGetTag

### PR DESCRIPTION
**Pull request by 6543**
`GET /repos/{owner}/{repo}/tags/{tag}` to get a single tag by name

### CODE REVIEW

File: routers/api/v1/repo/tag.go [hunk 1]
```diff
@@ -130,37 +130,22 @@ func DeleteTag(ctx *context.APIContext) {
 	//   required: true
 	// - name: tag
 	//   in: path
-	//   description: name of tag to delete
+	//   description: name of tag
 	//   type: string
 	//   required: true
 	// responses:
-	//   "204":
+	//   "200":
```
Comment (zeripath) [inline routers/api/v1/repo/tag.go:137]: You're not returning empty here

Comment (6543) [reply routers/api/v1/repo/tag.go:137]: No i return 200

Comment (6543) [reply routers/api/v1/repo/tag.go:137]: @zeripath DeleteTag codeblog got moved ...

Comment (zeripath) [reply routers/api/v1/repo/tag.go:137]: Look at the line below (L138) you tell swagger that you're returning /resposnses/empty

Comment (6543) [reply routers/api/v1/repo/tag.go:137]: haha right - good catch

**Review by zeripath — APPROVED** [review]

File: integrations/api_repo_tags_test.go [hunk 2]
```diff
@@ -51,6 +51,20 @@ func TestAPIRepoTags(t *testing.T) {
 			assert.EqualValues(t, newTag.Commit.SHA, tag.Commit.SHA)
 		}
 	}
+
+	// get created tag
+	req = NewRequestf(t, "GET", "/api/v1/repos/%s/%s/tags/%s?token=%s", user.Name, repoName, newTag.Name, token)
```
Comment (techknowlogick) [inline integrations/api_repo_tags_test.go:56]: Can you also add a test for a somewhat complicated tag, such as `gitea/123`?

**Comment by 6543** [comment]
@techknowlogick done

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**